lib/plugins/book.js

```diff
@@ -94,22 +97,24 @@ function inject (bot, { version }) {
         value: pages
       }
     }
-    bot.inventory.on('windowUpdate', onUpdate)
-    bot.inventory.updateSlot(slot, book)
-
-    function onUpdate (updatedSlot) {
-      if (updatedSlot === slot) {
-        bot.inventory.removeListener('windowUpdate', onUpdate)
-        cb(book)
+    return new Promise(resolve => {
+      function onUpdate (updatedSlot) {
+        if (updatedSlot === slot) {
+          bot.inventory.removeListener('windowUpdate', onUpdate)
+          resolve(book)
+        }
       }
-    }
-  }
 
-  bot.writeBook = (slot, pages, cb = noop) => {
-    write(slot, pages, null, null, false, cb)
+      bot.inventory.on('windowUpdate', onUpdate)
+      bot.inventory.updateSlot(slot, book)
+    })
```

Comment:
```js
bot.inventory.updateSlot(slot, book)
while (true) {
  const [updatedSlot] == await once(bot.inventory, 'windowUpdate')
  if (slot === updatedSlot) return
}
```

Reply:
Good one. it's only missing the return value which should be `book`. I think it should even be possible to do `await bot.inventory.once('windowUpdate')` instead.

Reply:
`await bot.inventory.once('windowUpdate')` does not do the same thing the once on the prototype of the EventEmiiter dosen't return a promise see https://nodejs.org/api/events.html#events_emitter_once_eventname_listener

Reply:
resolved in 43ba7aa